src/ripe/base.py

```diff
@@ -54,6 +54,127 @@ class API(
     transport_rule.TransportRuleAPI,
     availability_rule.AvailabilityRuleAPI,
 ):
+    @classmethod
+    def _query_to_spec(cls, query):
+        options = cls._unpack_query(query)
+        brand = options.get("brand", None)
+        model = options.get("model", None)
+        variant = options.get("variant", None)
+        version = options.get("version", None)
+        description = options.get("description", None)
+        initials = options.get("initials", None)
+        engraving = options.get("engraving", None)
+        gender = options.get("gender", None)
+        size = options.get("size", None)
+        meta = options.get("meta", [])
+        tuples = options.get("p", [])
+        tuples = tuples if isinstance(tuples, list) else [tuples]
+        initials_extra = options.get("initials_extra", [])
+        initials_extra = (
+            initials_extra if isinstance(initials_extra, list) else [initials_extra]
+        )
+        initials_extra = cls._parse_extra_s(initials_extra)
+        parts = cls._tuples_to_parts(tuples)
+        parts_m = cls._parts_to_parts_m(parts)
+        spec = dict(
+            brand=brand,
+            model=model,
+            parts=parts_m,
+            initials=initials,
+            engraving=engraving,
+            initials_extra=initials_extra,
+        )
+        if variant:
+            spec["variant"] = variant
+        if version:
+            spec["version"] = version
+        if description:
+            spec["description"] = description
+        if gender:
+            spec["gender"] = gender
+        if size:
+            spec["size"] = size
+        if meta:
+            spec["meta"] = cls._normalize_meta(meta)
+        return spec
+
+    @classmethod
+    def _unpack_query(cls, query):

```

Comment:
let's move all these methods to a util_api.py

Reply:
CC: @joao-conde

Reply:
Also didnt like these in the main class but didnt know what to call the new file since api was weird. util api it is :+1: